standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/model/MCreationMetadata.java

```diff
@@ -47,12 +75,13 @@ public MCreationMetadata(String catName, String dbName, String tblName,
     this.materializationTime = materializationTime;
   }
 
-  public Set<MTable> getTables() {
+  public Set<MMVSource> getTables() {
     return tables;
   }
 
-  public void setTables(Set<MTable> tables) {
-    this.tables = tables;
+  public void setTables(Set<MMVSource> tables) {
+    this.tables.clear();
```

Comment:
this change aims to make this class independent from the passed collection; I guess there were issues arising from it; but the constructor is still keeping the externally passed reference - if it caused problem maybe an  `ImmutableSet` would be better|?